### ISSUE TITLE
Shopify integation \\ ProductPicker shows only first 50 products.

### ISSUE THREAD

**Issue by SergeiPsarev** [umbraco/Umbraco.Cms.Integrations#157]
We've got hundreds of products in our Shopify but Umbraco.Cms.Integrations.Commerce.Shopify.ProductPicker allows to select from the first 50 products only (and ther is no pagination).

The issue is in the Umbraco.Cms.Integrations.Commerce.Shopify.Services.ShopifyService in GetResults() method:

      HttpRequestMessage request = new HttpRequestMessage()
      {
        Method = HttpMethod.Get,
        RequestUri = new Uri(string.Format("https://{0}.myshopify.com/admin/api/{1}/products.json", (object) this._settings.Shop, (object) this._settings.ApiVersion))
      };
            
Please read the following:
https://stackoverflow.com/questions/62272801/get-all-shopify-products
      
Basically if ?limit=XXX parameter is omitted in the RequestURI - Shopify returns first 50 products only. But setting limit to maximum 250 won't solve the problem either.
      
Ideal solution - implement pagination.
Quick fix solution - pull all products with several async API calls and merge the results. But that would make huge list of products in the ProductPicker.

      

---
_This item has been added to our backlog AB#36957_

**Comment by acoumb** [comment]
Hi @SergeiPsarev ,

And thank you for bringing this to our attention. I have released version [1.2.0](https://www.nuget.org/packages/Umbraco.Cms.Integrations.Commerce.Shopify/1.2.0) of the integration with pagination.

Regards,
Adrian
